compiler-rt/lib/mpk_untrusted/mpk_fault_handler.cpp

```diff
@@ -53,22 +53,22 @@ void disablePageMPK(siginfo_t *si, void *arg) {
 void disableThreadMPK(void *arg, uint32_t pkey) {
   uint32_t *pkru_ptr = __mpk_untrusted::pkru_ptr(arg);
 
-  last_pkey = pkey;
-  last_access_rights = __mpk_untrusted::pkey_get(pkru_ptr, pkey);
+  auto handler = __mpk_untrusted::AllocSiteHandler::get();
+  auto pkey_info = __mpk_untrusted::PKeyInfo(
+      pkey, __mpk_untrusted::pkey_get(pkru_ptr, pkey));
+  handler->storePidKey(getpid(), pkey_info);
   __mpk_untrusted::pkey_set(pkru_ptr, pkey, PKEY_ENABLE_ACCESS);
 
   __sanitizer::Report("INFO : Pkey(%d) has been set to ENABLE_ACCESS to enable "
                       "instruction access.\n",
                       pkey);
 }
 
-void enableThreadMPK(void *arg, uint32_t pkey) {
+void enableThreadMPK(void *arg, __mpk_untrusted::PKeyInfo pkey_info) {
   uint32_t *pkru_ptr = __mpk_untrusted::pkru_ptr(arg);
-  __mpk_untrusted::pkey_set(pkru_ptr, last_pkey, last_access_rights);
-  __sanitizer::Report("INFO : Pkey(%d) has been reset to %d.\n", last_pkey,
-                      last_access_rights);
-  last_pkey = INVALID_PKEY;
-  last_access_rights = PKEY_ENABLE_ACCESS;
+  __mpk_untrusted::pkey_set(pkru_ptr, pkey_info.pkey, pkey_info.access_rights);
+  __sanitizer::Report("INFO : Pkey(%d) has been reset to %d.\n", pkey_info.pkey,
```

Comment:
can we change these calls to `Report` to a macro that is only defined when a compile time flag is set? This can be in another PR, but is something we should address soon.

Reply:
Added a `REPORT` macro. Not sure if casing is important here (in terms of best practice), but I wanted to ensure I was distinct from the original Report function.

Reply:
All uppercase for preprocessor MACROS is normal